src/log_batch.rs

```diff
@@ -765,12 +767,62 @@ impl LogBatch {
     }
 
     #[allow(dead_code, unused_variables)]
-    pub(crate) fn sign_checksum(&mut self, file_version: Version, file_id: FileId) {
-        // @TODO: lucasliang
-        // Supplement the details of signing checksum on each LogBatch with the
-        // specific `Version`.
-        unimplemented!();
-        // debug_assert!(matches!(self.buf_state, BufState::Sealed(_, _)));
+    /// Signs a checksum, so-called `signature`, into the `LogBatch`.
+    ///
+    /// The `signature` is both generated by the given `[Version]` and
+    /// `[FileId]`. That is, the final checksum of each `LogBatch`
+    /// consists of this `signature` and the `checksum` of the contents.
+    pub(crate) fn sign_checksum(&mut self, format_version: Version, file_id: FileId) {
+        debug_assert!(matches!(self.buf_state, BufState::Sealed(_, _)));
+        if !self.is_empty() {
+            match format_version {
```

Comment:
Put this logic into a new method of `LogItemBatch`. Because the un-sign-ing is done in item batch.

Reply:
Copy that.